contrib/babelfishpg_tsql/src/session.c

```diff
@@ -376,3 +394,54 @@ initialize_context_table()
 
 	session_context_table = hash_create("Session Context", 128, &hash_options, HASH_ELEM | HASH_STRINGS);
 }
+
+/* 
+* This function is responsible for estimating the size of the entry and the number of keys 
+* and insert into the DSM for parallel workers
+* The first argument is ParallelContext which contains the info related to TOC
+* The second argument indicates whether we want to estimate the space or
+* we want to insert the data into DSM
+*/
+void
+babelfixedparallelstate_insert(ParallelContext *pcxt, bool estimate)
+{
+	BabelfishFixedParallelState *bfps;
+	int len;
+	char* current_db_name;
+	if (estimate)
+	{
+		/* Allow space to store the babelfish fixed-size parallel state. */
+		shm_toc_estimate_chunk(&pcxt->estimator, sizeof(BabelfishFixedParallelState));
+		shm_toc_estimate_keys(&pcxt->estimator, 1);
+	}
+	else
+	{
+		/* Initialize babelfish fixed-size state in shared memory. */
+		bfps = (BabelfishFixedParallelState *) shm_toc_allocate(pcxt->toc, sizeof(BabelfishFixedParallelState));
+		current_db_name = get_cur_db_name();
+
+		if(current_db_name == NULL || !DbidIsValid(get_db_id(current_db_name)))
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_DATABASE),
+					errmsg("database \"%s\" does not exist", current_db_name)));
```

Comment:
I think %s with NULL string would be an issue. You can use "" in case of NULL

Reply:
Updated